torch/utils/mem_tracker.py

```diff
@@ -0,0 +1,242 @@
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Any, Dict
+import torch
+import math
+from torch._subclasses.fake_tensor import FakeTensorMode
+from torch.utils._pytree import tree_map_only
+import torchvision
+import torch.nn as nn
+from torch.utils._python_dispatch import TorchDispatchMode
+from torch.utils.weak import WeakIdKeyDictionary
+import weakref
+from enum import Enum
+
+MB = 2**20
+KB = 2**10
+MEMORY_MAX = 0
+PYTORCH_MIN_ALLOCATE = 2**9
+MEMORY_USE = WeakIdKeyDictionary()
```

Comment:
I think now that we're making this a real thing, we should remove this global and just create it on entry to the mode.

Reply:
Done